src/test/java/com/scottlogic/sorts/NameDatePostSorterTest.java

```diff
@@ -0,0 +1,119 @@
+
+        List<UserPost> actual = new NameDatePostSorter(localDatePostSorter, localAuthorPostSorter)
+                .sort(null, null);
+
+        assertNull(actual);
+    }
+
+    @Test
+    void sort_EmptyList_EmptyList() {
+        List<UserPost> inputList = List.of();
+        List<UserPost> expected = List.of();
+
+        when(localAuthorPostSorter.sort(inputList, SortOrder.ASC)).thenReturn(inputList);
+        when(localDatePostSorter.sort(inputList, SortOrder.ASC)).thenReturn(inputList);
+
+        List<UserPost> actual = new NameDatePostSorter(localDatePostSorter, localAuthorPostSorter)
+                .sort(inputList, SortOrder.ASC);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void sort_1ItemList_1ItemList() {
+        List<UserPost> inputList = Arrays.asList(Smooth2020_01_01_0230);
+        List<UserPost> expected = List.of(Smooth2020_01_01_0230);
+
+        when(localAuthorPostSorter.sort(inputList, SortOrder.ASC)).thenReturn(inputList);
+        when(localDatePostSorter.sort(inputList, SortOrder.ASC)).thenReturn(inputList);
+
+        List<UserPost> actual = new NameDatePostSorter(localDatePostSorter, localAuthorPostSorter)
+                .sort(inputList, SortOrder.ASC);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void sort_2ItemsWrongOrder_2ItemsRightOrder() {
+        List<UserPost> inputList = List.of(Smooth2020_01_01_0230, Biggs2020_01_01_0230_of2);
+        List<UserPost> expected = List.of(Biggs2020_01_01_0230_of2, Smooth2020_01_01_0230);
+
+        when(localDatePostSorter.sort(inputList, SortOrder.ASC))
+                .thenReturn(List.of(Smooth2020_01_01_0230, Biggs2020_01_01_0230_of2));
+        when(localAuthorPostSorter.sort(List.of(Smooth2020_01_01_0230, Biggs2020_01_01_0230_of2), SortOrder.ASC))
+                .thenReturn(List.of(Biggs2020_01_01_0230_of2, Smooth2020_01_01_0230));
+
+        List<UserPost> actual = new NameDatePostSorter(localDatePostSorter, localAuthorPostSorter)
+                .sort(inputList, SortOrder.ASC);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void sort_3ItemsUnSorted_3ItemsSorted() {
+        List<UserPost> inputList = List.of(Smooth2020_02_01_2200, Smooth2020_01_01_0230, Small2021_01_01_1230);
+        List<UserPost> expected = List.of(Small2021_01_01_1230, Smooth2020_01_01_0630, Smooth2020_02_01_2200);
+
+        when(localDatePostSorter.sort(inputList, SortOrder.ASC))
+                .thenReturn(List.of(Smooth2020_01_01_0630, Smooth2020_02_01_2200, Small2021_01_01_1230));
+        when(localAuthorPostSorter.sort(List.of(Smooth2020_01_01_0630, Smooth2020_02_01_2200, Small2021_01_01_1230), SortOrder.ASC))
+                .thenReturn(List.of(Small2021_01_01_1230, Smooth2020_01_01_0630, Smooth2020_02_01_2200));
+
+        List<UserPost> actual = new NameDatePostSorter(localDatePostSorter, localAuthorPostSorter)
+                .sort(inputList, SortOrder.ASC);
+
+        assertEquals(expected, actual);
+    }
```

Comment:
Probably worth testing some that use `SortOrder.Desc` as well as this is definitely a case that could cause some strange situations